src/include/column_inline.h

```diff
@@ -56,9 +63,20 @@ __col_insert_search_gt(WT_INSERT_HEAD *ins_head, uint64_t recno)
      * There isn't any safety testing because we confirmed such a record exists before searching.
      */
     if ((ins = ret_ins) == NULL)
-        ins = WT_SKIP_FIRST(ins_head);
+        /*
+         * CPU may reorder the reads and return a stale value of the head.
+         *
+         * Place a read barrier here to avoid this issue.
+         */
+        WT_ORDERED_READ(ins, WT_SKIP_FIRST(ins_head));
     while (recno >= WT_INSERT_RECNO(ins))
-        ins = WT_SKIP_NEXT(ins);
+        /*
+         * CPU may reorder the read and we may read a stale next value and incorrectly skip that
+         * key.
+         *
+         * Place a read barrier to avoid this issue.
+         */
+        WT_ORDERED_READ(ins, WT_SKIP_NEXT(ins));
```

Comment:
I think I'm misunderstanding something about this section. It's fine to keep traversing along a single level even if we're potentially missing an entry, right? Since the old pointer will still be valid. It might just look like we're missing an entry, which I don't think we can do much about (sure, we're reading in order, but some other thread might have modified it and not flushed out the write yet.)

Reply:
No, it is not if we have seen the new insert list upper the stack.

Reply:
Oh I think I see - is it that we're protecting against a reordered read from the previous section where we're descending down the levels?

Reply:
Yes, that's correct.